sql/src/main/java/io/crate/operation/auth/AuthenticationMethod.java

```diff
@@ -24,18 +24,20 @@
 
 import io.crate.operation.user.User;
 import io.crate.protocols.postgres.ConnectionProperties;
+import org.elasticsearch.common.settings.SecureString;
 
 import javax.annotation.Nullable;
 
 public interface AuthenticationMethod {
 
     /**
      * @param userName the userName sent with the startup message
+     * @param passwd the password in clear-text or null
      * @return the user or null; null should be handled as if it's a "guest" user
      * @throws RuntimeException if the authentication failed
      */
     @Nullable
-    User authenticate(String userName, ConnectionProperties connProperties);
+    User authenticate(String userName, @Nullable SecureString passwd, ConnectionProperties connProperties);
```

Comment:
Have you considered making the password part of the `ConnectionProperties`, maybe rename it to AuthenticationProperties? The password is an optional parameter, much like the ssl options in ConnectionProperties.

Reply:
yes, I also thought about that.
but then also the username would need to be part of the connection properties class.
also, the authentication context is only needed for the postgres protocol implementation, and would probably not be required for the HTTP implementation